File: frontend/src/types/pool.ts
import { MemberResponse } from './auth';
import { TeamSummary } from './scoreboard';

export interface SeasonAssignmentResponse {
  id: number;
  season_year: number;
  member: MemberResponse;
  team: TeamSummary;
}

export interface WinningGameResponse {
  id: number;
  season_year: number;
  week: number;
  winnings: number;
  winning_type: string;
  team: TeamSummary;
  winning_owner_name: string;
}

export interface StandingItem {
  member: MemberResponse;
  team: TeamSummary;
  total_winnings: number;
}

export interface PoolResultsResponse {
  season_year: number;
  winning_games: WinningGameResponse[];
  standings: StandingItem[];
}

export interface PotResponse {
  season_year: number;
  amount: number;
}

export interface SeasonsResponse {
  current_season: number;
  tracked_seasons: number[];
}

File: apps/football_pool/schemas/pool.py
"""Schemas for pool assignments, results, standings, and pot."""

from __future__ import annotations

from pydantic import BaseModel

from .auth import MemberResponse
from .scoreboard import TeamSummary


class SeasonAssignmentResponse(BaseModel):
    """A member's team assignment for a specific season."""

    id: int
    season_year: int
    member: MemberResponse
    team: TeamSummary


class WinningGameResponse(BaseModel):
    """Historical payout record for a team in a given week."""

    id: int
    season_year: int
    week: int
    winnings: int
    winning_type: str
    team: TeamSummary
    winning_owner_name: str


class StandingItem(BaseModel):
    """Leaderboard entry showing a member's dynamic winnings for a season."""

    member: MemberResponse
    team: TeamSummary
    total_winnings: int


class PoolResultsResponse(BaseModel):
    """Aggregate results response containing winning games and member standings."""

    season_year: int
    winning_games: list[WinningGameResponse]
    standings: list[StandingItem]


class PotResponse(BaseModel):
    """Current pot amount for a season."""

    season_year: int
    amount: int


class SeasonsResponse(BaseModel):
    """Listing of tracked seasons."""

    current_season: int
    tracked_seasons: list[int]
